projects/smoke_ltc6811/src/main.c

```diff
@@ -0,0 +1,200 @@
+#include <string.h>
+
+#include "critical_section.h"
+#include "event_queue.h"
+#include "interrupt.h"
+#include "log.h"
+#include "ltc_afe.h"
+#include "soft_timer.h"
+#include "status.h"
+
+// device settings
+#define SMOKE_LTC_AFE_NUM_DEVICES 1
+#define SMOKE_LTC_AFE_NUM_CELLS 12
+#define SMOKE_LTC_AFE_INPUT_BITSET_FULL 0xFFF
+
+#define SMOKE_LTC_AFE_ADC_MODE LTC_AFE_ADC_MODE_7KHZ
+
+#define SMOKE_LTC_AFE_SPI_PORT SPI_PORT_1
+#define SMOKE_LTC_AFE_SPI_BAUDRATE 750000
+#define SMOKE_LTC_AFE_SPI_MOSI \
+  { .port = GPIO_PORT_A, .pin = 7 }
+#define SMOKE_LTC_AFE_SPI_MISO \
+  { .port = GPIO_PORT_A, .pin = 6 }
+#define SMOKE_LTC_AFE_SPI_SCLK \
+  { .port = GPIO_PORT_A, .pin = 5 }
+#define SMOKE_LTC_AFE_SPI_CS \
+  { .port = GPIO_PORT_A, .pin = 4 }
+
+// processing events
+typedef enum {
+  SMOKE_LTC_AFE_TRIGGER_CELL_CONV_EVENT = 0,
+  SMOKE_LTC_AFE_CELL_CONV_COMPLETE_EVENT,
+  SMOKE_LTC_AFE_TRIGGER_AUX_CONV_EVENT,
+  SMOKE_LTC_AFE_AUX_CONV_COMPLETE_EVENT,
+  SMOKE_LTC_AFE_CALLBACK_RUN_EVENT,
+  SMOKE_LTC_AFE_FAULT_EVENT,
+  NUM_SMOKE_LTC_EVENTS,
+} TestLtcAfeEvents;
+
+#define SMOKE_LTC_SHOULD_LOG_EVENTS true
```

Comment:
could we put this at the very top so it's easier for a hardware person to notice it? (maybe right after the includes) Also since there could be a lot going on, could you add #defines for cell conv vs. aux conv so we could debug either one or the other then only log depending on what #defines are set?